Heffner_Daniel/Rails Section 2/full_secrets_app/app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SessionsController < ApplicationController
+
+	def new
+		@session = User.new
+	end
+
+	def create
+		@session = User.find_by( email: user_params[:email] )
+		if @session
+			if !@session.authenticate(user_params[:password])
+				flash[:alert] = "Invalid"
+				redirect_to new_session_path
+			else
+				session[:current_user] = User.select(:id, :name, :email).where(id: @session.id)[0]
```

Comment:
You have @session, so no need to go back to the db. You can possibly do @session.slice(:id, :name, :email)